apps/authentication-service/src/shared/tokenUtils.ts

```diff
@@ -0,0 +1,69 @@
+import fs from "fs";
+import { decode, JwtPayload } from "jsonwebtoken";
+import os from "os";
+import path from "path";
+
+import { ConsoleColors } from "@/shared/enums";
+
+let token: string | undefined;
+const tokenFileName: string = "brkn-arrw.txt";
+
+export const tokenFile: string = path.join(os.tmpdir(), tokenFileName);
+
+export const getLocalToken = (): string | undefined => {
+  logToken(token);
+  return token;
+};
+
+export const setLocalToken = (newToken: string | undefined): void => {
+  token = newToken;
+};
+
+export const setCachedToken = (token: string): boolean => {
+  try {
+    fs.writeFileSync(tokenFile, token);
+    return true;
+  } catch (error) {
+    return false;
+  }
+};
+
+export const getCachedToken = (): string | undefined => {
+  if (fs.existsSync(tokenFile)) {
+    return fs.readFileSync(tokenFile, "utf8");
+  } else {
+    return undefined;
+  }
+};
+
+export const getIsTokenExpired = (
+  token: string | undefined | null,
+): boolean => {
+  if (!token) {
+    return true;
+  }
+
+  try {
+    const { exp } = decode(token, { json: true }) as JwtPayload;
+    return !exp || Date.now() >= exp * 1000;
+  } catch (Error) {
+    return true;
+  }
+};
+
+export const logToken = (token: string | undefined) => {
+  if (
+    !token ||
+    (process.env.JEST_WORKER_ID !== undefined && !process.env.JEST_NO_SKIP)
+  ) {
+    return;
+  }
+
+  if (process.env.ENV === "test") {
+    console.log(`\nToken: ${ConsoleColors.FgGray}${token}`);
```

Comment:
I don't think we should ever be logging with the token. This could get exposed in our end to end tests.